### PR TITLE
Wos xml parser

### PR DESCRIPTION
**Pull request by darrenleeweber**
Extracted from #379 

This greatly simplifies the WOS-record(s) parsing and extraction for XML data.  It decreates the memory footprint for those classes too, by hanging onto only the data it needs.

### CODE REVIEW

File: spec/lib/web_of_science/xml_parser_spec.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
+require 'htmlentities'
+
+describe WebOfScience::XmlParser do
+  let(:wos_encoded_record) { File.read('spec/fixtures/wos_client/wos_encoded_record.html') }
+  let(:wos_record_encoded) { WebOfScience::Record.new(encoded_record: wos_encoded_record) }
+
+  shared_examples 'it has well formed XML' do
+    let(:html_char) { '&lt;' }
```
Comment (atz) [inline spec/lib/web_of_science/xml_parser_spec.rb]: This is only used once.  It doesn't seem to need `let` semantics.

Comment (darrenleeweber) [reply spec/lib/web_of_science/xml_parser_spec.rb]: OK, removed

File: spec/lib/web_of_science/xml_parser_spec.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    end
+  end
+
+  describe '#to_xml' do
+    let(:xml_result) { wos_record_encoded.to_xml }
+
+    it_behaves_like 'it has well formed XML'
+  end
+
+  describe '#parse_xml' do
```
Comment (atz) [inline spec/lib/web_of_science/xml_parser_spec.rb]: The method being tested here is `#parse`

Comment (darrenleeweber) [reply spec/lib/web_of_science/xml_parser_spec.rb]: Done

**Pull request merged**